Makefile

```
@@ -23,28 +23,28 @@ else ifeq ($(shell uname -s),Darwin)
 endif
 
 pin-toolchain-mac-m1:
-	@rustup toolchain install nightly-2023-03-14 --component llvm-tools-preview
-	@rustup target add wasm32-unknown-unknown --toolchain nightly-2023-03-14
+	@rustup toolchain install nightly-2023-04-21 --component llvm-tools-preview
+	@rustup target add wasm32-unknown-unknown --toolchain nightly-2023-04-21
 	@rm -rf ~/.rustup/toolchains/nightly-aarch64-apple-darwin
-	@ln -s ~/.rustup/toolchains/nightly-2023-03-14-aarch64-apple-darwin ~/.rustup/toolchains/nightly-aarch64-apple-darwin
+	@ln -s ~/.rustup/toolchains/nightly-2023-04-21-aarch64-apple-darwin ~/.rustup/toolchains/nightly-aarch64-apple-darwin
 
 pin-toolchain-linux:
-	@rustup toolchain install nightly-2023-03-14 --component llvm-tools-preview
-	@rustup target add wasm32-unknown-unknown --toolchain nightly-2023-03-14
+	@rustup toolchain install nightly-2023-04-21 --component llvm-tools-preview
+	@rustup target add wasm32-unknown-unknown --toolchain nightly-2023-04-21
 	@rm -rf ~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu
-	@ln -s ~/.rustup/toolchains/nightly-2023-03-14-x86_64-unknown-linux-gnu ~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu
+	@ln -s ~/.rustup/toolchains/nightly-2023-04-21-x86_64-unknown-linux-gnu ~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu
 	@rustup component add clippy --toolchain nightly-x86_64-unknown-linux-gnu
 
 lint:
 	@echo ⚙️ Running the linter...
-	@cargo +nightly clippy --workspace --all-targets -- -D warnings
+	@cargo +nightly clippy --workspace -Fbinary-vendor --all-targets -- -D warnings
 
 pre-commit: fmt lint full-test
 
 test:
 	@echo ⚙️ Running unit tests...
-	@cargo +nightly t
+	@cargo +nightly t --release
 
 full-test:
 	@echo ⚙️ Running all tests...
-	@cargo +nightly t -- --include-ignored --test-threads=1
+	@cargo +nightly t --release -Fbinary-vendor --  --include-ignored --test-threads=1
```

```suggestion
	@cargo +nightly t -Fbinary-vendor --  --include-ignored --test-threads=1
```
Tests should run in the debug build.